src/Plugin.ts

```diff
@@ -99,8 +100,19 @@ export class Plugin {
     );
 
     await networkObservable.subscribe(
-      async (request: NetworkRequest): Promise<number> =>
-        this.entries.push(await new EntryBuilder(request).build())
+      async (request: NetworkRequest): Promise<number> => {
+        const requestPath = request.parsedURL.path;
+        const isRequestExcluded = options.excludePaths?.some(
+          (excludedPath: string): boolean =>
+            requestPath?.startsWith(excludedPath)
+        );
+
+        if (isRequestExcluded) {
+          return this.entries.length;
+        }
```

Comment:
I suggest you move this logic to `NetworkObservale`.  You should prevent creating requests in `webSocketCreated` and `requestWillBeSent` handlers.